linode/py-metadata
# Implement Sync and Async Watcher

## 📝 Description

This PR is for adding the watchers.

## ✔️ How to Test

### Automated Testing
```bash
make e2e
```

### Manual Testing (in a Linode instance)

- Run this program:

Async version:
```python
import asyncio
from datetime import timedelta
from linode_metadata import AsyncMetadataClient

async def get_metadata():
    async with AsyncMetadataClient() as client:
        watcher = client.get_watcher()
        async for new_network_info in watcher.watch_network(timedelta(seconds=5)):
            print(new_network_info)

asyncio.run(get_metadata())
```

Sync version:
```python
from datetime import timedelta
from linode_metadata import MetadataClient

def get_metadata():
    with MetadataClient() as client:
        watcher = client.get_watcher()
        for new_network_info in watcher.watch_network(timedelta(seconds=5)):
            print(new_network_info)

get_metadata()
```
- You should observe there is one network info printed from the shell.
- Edit the networking config of the instance, e.g. adding one private IPv4 address.
- You should observe a new network info printed from the shell, which contains the newly created IPv4 address.
